add fastq cleanup to sra path

jodyphelan opened the pull request:
Sometimes there are singleton reads in the forward or reverse read file. This creates issues with mismatching read names that causes `BWA` to fail. This adds an extra step to extract only reads which appear in both forwards and reverse fastq files.

jodyphelan committed:
add fastq cleanup to sra path

sachalau commented on containers/seqkit/Dockerfile:
```diff
@@ -0,0 +1,14 @@
+
+ADD cleanup_normalized_fastq.sh /scripts/
+
+WORKDIR /scratch
```
You'll need to create a new repository in AWS ECR so that the new image can be stored. This is controlled in the tbsequencing-infrastructure repository, ecr.tf file.

jodyphelan replied:
I've added a PR for this here: https://github.com/WorldHealthOrganization/tbsequencing-infrastructure/pull/2

sachalau commented on devops/envs/prod-unicc/pipeline_child.json:
```diff
@@ -364,6 +364,39 @@
+                                "MaxAttempts": 10,
+                                "BackoffRate": 2
+                            }
+                        ],
```
${SeqKit} variable is undefined. You need to create a new batch job definition in the batch.tf file and then pass the arn for batch job definition via the ${SeqKit} variable.

jodyphelan replied:
Ok will do.

sachalau commented on devops/envs/prod-unicc/pipeline_child.json:
```diff
@@ -364,6 +364,39 @@
+                                ]
+                            },
+                            "Parameters": {
+                                "SAMPLE_ID.$": "States.Format('{}', $.SampleId)",
```
don't think there's a need to use States.Format here ?

jodyphelan replied:
I wasn't sure as it is used like this in other parameter definitions, e.g.: https://github.com/WorldHealthOrganization/tbsequencing-bioinfoanalysis/blob/030bd6cb0279d114ed68e8153d5be3fad8974aeb/devops/envs/prod-unicc/pipeline_child.json#L353

I'm not too familiar with step function job definitions, can it be simplified to `"SAMPLE_ID.$": "$.SampleId"`?

sachalau replied:
Oh no I get it now. $.SampleId is an integer, it must be transformed to string. Ignore my comment.

sachalau replied:
revert the change sorry :(

jodyphelan replied:
No probs - just updated now!

sachalau commented on containers/seqkit/cleanup_normalized_fastq.sh:
```diff
@@ -0,0 +1,21 @@
+SAMPLE_ID="$1"
+FASTQ_ID="$2"
+
+cd "$SAMPLE_ID"
```
are you sure you need to cd ? all other command include the $SAMPLE_ID param at the beginning of the file paths.

jodyphelan replied:
Oh yeah I'll remove that

sachalau reviewed (changes requested):
You'll also need to add a new step in the GitHub CI to build the seqkit image and push it to ECR (it's in .github/workflows/push.yml)

jodyphelan committed:
add seqkit ARN in batch.tf

jodyphelan committed:
remove redundant cd command

jodyphelan committed:
add seqkit to github actions push.yml

jodyphelan committed:
add seqkit to github actions push.yml

jodyphelan commented:
I've made the following changes:
* Added seqkit to github actions workflow push.yml
* Removed redundant cd in fastq cleanup script
* Added seqkit ARN in batch.tf
* Added a PR to add seqkit to ecr.tf in github.com/WorldHealthOrganization/tbsequencing-infrastructure/

sachalau commented:
Last thing you need to do is in stepfunctions.tf, for the child pipeline definition, pass a Seqkit variable which points to the arn of the batch job definition output you created.

jodyphelan committed:
add seqkit to step_functions.tf

jodyphelan commented:
Ok I've added this now

sachalau commented:
Sorry you also need to revert one of the change I requested, I commented again about it above.

jodyphelan committed:
update state format

jodyphelan commented:
No probs, made the change back now.

Commit:
terraform format
